src/commands/generate/types.ts

```diff
@@ -0,0 +1,123 @@
+import { CSharpFileGenerator, JavaFileGenerator, JavaScriptFileGenerator, TypeScriptFileGenerator, GoFileGenerator, Logger, DartFileGenerator} from '@asyncapi/modelina';
+import { Flags } from '@oclif/core';
+import Command from '../../base';
+import { load } from '../../models/SpecificationFile';
+import { parse } from '@asyncapi/parser';
+enum Languages {
+  typescript = 'typescript',
+  csharp = 'csharp',
+  golang = 'golang',
+  java = 'java',
+  javascript = 'javascript',
+  dart = 'dart'
+}
+export default class Types extends Command {
+  static description = 'Generates typed models';
+
+  static args = [
+    { 
+      name: 'language', 
+      description: 'language output', 
+      options: Object.keys(Languages), 
+      required: true 
+    },
+    { name: 'file', description: 'spec path, url, or context-name', required: true },
+  ]
+
+  static flags = {
+    help: Flags.help({ char: 'h' }),
+    output: Flags.string({ char: 'o', description: 'output path to write the model to' }),
```

Comment:
If this is going to be a directory, should we call it `outputDir` instead? Or is this flag representing the name of a single file?

Reply:
It could be `outputDir` yes, that would be more precise as it is the directory where all the models are going to be generated to 🤔

Reply:
I think it's just a matter of what I found most common in my career. `output` usually refers to a file or stdout. When the output is a whole dir, it should be noted IMHO. I don't think many people will expect the output to be a dir, to be honest. Anyway, not a blocker for me, just my two cents.